tests/schema/tests.py

```diff
@@ -1745,6 +1745,47 @@ class Meta:
             editor.alter_field(LocalBook, old_field, new_field, strict=True)
         self.assertForeignKeyExists(LocalBook, "author_id", "schema_author")
 
+    @skipUnlessDBFeature("supports_foreign_keys")
+    def test_fk_constraint_index_drop(self):
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+            editor.create_model(Book)
+
+        author = Author.objects.create(name="Alice")
+        Book.objects.create(
+            author=author,
+            title="Much Ado About Foreign Keys",
+            pub_date=datetime.datetime.now(),
+        )
+
+        old_field = Book._meta.get_field("author")
+        new_field = ForeignKey(Author, CASCADE, db_index=False)
+        new_field.set_attributes_from_name("author")
+        with (
+            CaptureQueriesContext(connection) as ctx,
+            connection.schema_editor() as editor,
+        ):
+            editor.alter_field(Book, old_field, new_field, strict=True)
+
+        if connection.features.requires_fk_constraints_to_be_recreated:
+            self.assertIs(
+                any(
+                    "DROP FOREIGN KEY" in query["sql"] for query in ctx.captured_queries
+                ),
+                True,
+            )
```

Comment:
Since [`any` always returns a `bool`](https://docs.python.org/3/library/functions.html#any), I'd use the simpler to read `assertTrue`:

```suggestion
            self.assertTrue(
                any(
                    "DROP FOREIGN KEY" in query["sql"] for query in ctx.captured_queries
                )
            )
```